lib/improver/database/spotdata_database.py

```diff
@@ -0,0 +1,276 @@
+                except AttributeError:
+                    self.df = df
+
+        #self.df = self.df.pivot(columns = 'forecast_period', values='vals')
+
+    def constrained_to_dataframe():
+        """
+        Turn the input cubes into a dataframe sorted by validity_time
+
+        """
+        cubes = self.cubelist
+        rows = self.primary_dim
+        coords = [coord.standard_name or coord.long_name for coord in
+                                                    self.cubelist[0].coords()]
+        for cube in cubes:
+            ignored_coord_dims = [item for item in ignored_coords if item not in cube.dim_coords]
+            print cube
+            for c in cube.slices_over('time'):
+                df = iris.pandas.as_data_frame(c, copy=False)
+
+        primary_key_iterator = self.determine_range(rows)
+        for row_val in sorted(primary_key_iterator, key=lambda x: x.points):
+            print row_val
+            records = []
+
+            # Constrain the cubes by primary_dim
+            constraint = iris.Constraint(**{rows: row_val.points})
+            selection = self.cubelist.extract(constraint)
+
+            record = dict()
+            for cube in selection:
+                df = iris.pandas.as_data_frame(cube, copy=False)
+            return
+            for iterator in iris.iterate.izip(*selection):
+                df = iris.pandas.as_data_frame(cube, copy=False)
+                for cube in iterator:
+                    dictionary = cube.coords()
+                    record = {coord.standard_name or
+                              coord.long_name: coord.points[0]
+                              for coord in [cube.coord(c) for c in cols]}
+                    records.append(record)
+            new_df = DataFrame.from_dict(records)
+            self.df = pd.concat([self.df, new_df])
+        return self.df
+
+    def create_table(self, outfile, table='test'):
+        """
+        Create the SQL datafile table
+
+        """
+
+        if os.path.isfile(outfile):
+            os.unlink(outfile)
+
+        # Remove the current index, and use the indexed columns for for db keys
+        columns = self.df.columns
+        new_df = self.df.reset_index()
+        n_keys = len(new_df) - len(columns)
+
+        schema = pd.io.sql.get_schema(self.df, table,
+                                      flavor='sqlite',
+                                      keys=self.df.columns[:n_keys])
+
+        with sqlite3.connect(outfile) as db:
+            db.execute(schema)
+
+    def to_sql(self, outfile, table='test', new=True):
+        """
+        Output the dataframe to SQL database file
+
+        """
+
+        with sqlite3.connect(outfile) as db:
+            self.df.to_sql(table, con=db, if_exists='append', index=True)
+
+    def to_csv(self, outfile):
+        """
+        Output the dataframe to comma seperated file
+
+        """
+
+        self.df.to_csv(outfile)
+
+    def determine_range(self, dimension):
```

Comment:
I don't think you need this function. By definition an iris coordinate is monotonically increasing and does not allow for duplicate values in its points. If you still want to make sure you can do this in one line:
`unique_values =  set(cube.coord(dimension).points)`

Reply:
I don't think so, will remove